charts/openstack-cluster/templates/node-group/kubeadm-config-template.yaml

```diff
@@ -19,13 +19,26 @@
 joinConfiguration: {{ include "openstack-cluster.nodeRegistration.nodeLabels" $nodeLabels | nindent 2 }}
 {{- end }}
 
+{{- define "openstack-cluster.nodegroup.kubeadmConfigSpec.containerdRestart" -}}
+{{- $ctx := index . 0 }}
+{{- $nodeGroup := index . 1 }}
+{{- $proxy := $ctx.Values.proxy | default dict }}
+{{- if and (hasKey $proxy "http_proxy") (hasKey $proxy "https_proxy") (hasKey $proxy "no_proxy") }}
+preKubeadmCommands:
+  - systemctl daemon-reload
+  - systemctl restart containerd
+{{- end }}
+{{- end }}
+
+
```

Comment:
This is no longer required if `preKubeadmCommands` are added to `openstack-cluster.kubeadmConfigSpec.containerd`.